### PR TITLE
gRPC blanket implementation from ibc-rs

### PR DESCRIPTION
**Pull request by rnbguy**
This PR uses gRPC blanket implementation from [ibc-rs](https://github.com/cosmos/ibc-rs/pull/842) using `QueryContext` and `ProvableContext` traits.

Todo:
- [x] Merge `ibc-rs` PR and update commit rev in this PR
- [x] Update Cargo.lock
- [x] Remove old `service.rs`
- [x] Add comments
- [x] Add more `grpcurl` tests
- [x] Fix `client_status` endpoint
- [x] Add unclog entry

### CODE REVIEW

File: crates/app/src/modules/ibc/impls.rs [hunk 1]
```diff
@@ -147,16 +154,24 @@ where
...
     }
 
-    pub fn client_service(&self) -> ClientQueryServer<IbcClientService<S>> {
-        ClientQueryServer::new(IbcClientService::new(self.ctx.store.clone()))
+    pub fn client_service(
+        &self,
+        update_context: &Upgrade<S>,
+    ) -> ClientQueryServer<ClientQueryService<IbcContext<S>, Upgrade<S>>> {
+        ClientQueryServer::new(ClientQueryService::new(
+            self.ctx.clone(),
```
[the remark this answers is not in the document]
Comment (rnbguy) [reply crates/app/src/modules/ibc/impls.rs:162]: Very good catch. As commented on ibc-rs PR, @Farhad-Shabani suggested making `IbcContext` to be `Sync`.

[the remark this answers is not in the document]
Comment (Farhad-Shabani) [reply crates/app/src/modules/ibc/impls.rs:162]: https://github.com/cosmos/ibc-rs/pull/842#discussion_r1323181692

File: Cargo.toml [hunk 2]
```diff
@@ -12,7 +12,7 @@ base64 = { version = "0.21", default-features = false, features = ["alloc"] }
 displaydoc = { version = "0.2", default-features = false }
 derive_more = { version = "0.99.17", default-features = false, features = ["from", "into", "display"] }
 ed25519 = { version = "2.1.0", default-features = false }
-ibc = { git = "https://github.com/cosmos/ibc-rs.git", rev= "6ffe8ce" }
+ibc = { git = "https://github.com/cosmos/ibc-rs.git", rev= "75ca4ca", features = ["std", "grpc"] }
```
Comment (Farhad-Shabani) [inline Cargo.toml:15]:
```suggestion
ibc = { git = "https://github.com/cosmos/ibc-rs.git", rev= "75ca4ca", features = ["grpc"] }
```

File: crates/app/src/modules/ibc/impls.rs [hunk 3]
```diff
@@ -147,16 +149,24 @@ where
...
+            update_context.clone(),
+        ))
```
Comment (Farhad-Shabani) [inline crates/app/src/modules/ibc/impls.rs:159]:
```suggestion
    pub fn client_service(
        &self,
        upgrade_context: &Upgrade<S>,
    ) -> ClientQueryServer<ClientQueryService<IbcContext<S>, Upgrade<S>>> {
        ClientQueryServer::new(ClientQueryService::new(
            self.ctx.clone(),
            upgrade_context.clone(),
        ))
```

**Comment by Farhad-Shabani** [comment]
> Currently, `ibc_context.clone()` will not create fields with _shared ownership_, as some fields are not wrapped in `Arc<_>`.

Much needed improvement! This has caused me trouble in a few spots. Each time, I got caught up with other urgent tasks and didn't get around to update it.

> Considering this, I was wondering if the fix should be in this PR or a separate one.
> WDYT?

I'd suggest in a separate PR.

**Comment by rnbguy** [comment]
Looks like `client_status` calls [`host_consensus_state()`](https://github.com/informalsystems/basecoin-rs/blob/29ff4cf41424ac481dcd8c174de5e91189147bc5/crates/app/src/modules/ibc/impls.rs#L427) which access one of those fields which are not wrapped in `Arc<_>` :sweat_smile: and it threw error as expected when that endpoint is queried.

I will wrap only that field to keep this PR wokring. The `Arc<_>` wrap for other fields will be on future PR.

**Review by Farhad-Shabani — APPROVED** [review]

**Pull request merged**